apis/v1alpha1/httproute_types.go

```diff
@@ -76,7 +85,24 @@ type HTTPRouteHost struct {
 	// Support: custom
 	//
 	// +optional
-	ExtensionRef *RouteHostExtensionObjectReference `json:"extensionRef" protobuf:"bytes,3,opt,name=extensionRef"`
+	ExtensionRef *RouteHostExtensionObjectReference `json:"extensionRef" protobuf:"bytes,4,opt,name=extensionRef"`
+}
+
+// TLSinRouteConfig describes a TLS configuration defined at the Route level.
+type TLSInRouteConfig struct {
```

Comment:
Does it make sense to define this in tlsconfig_types.go ?

Reply:
That's a good point. If we end up having separate TLSConfig types per resource maybe we can leave this here? For BackendTLSConfig I'm proposing keeping it in the `backendpolicy_types.go` file. On that note, would `RouteTLSConfig` or `HTTPRouteTLSConfig` be a better name here? I'm not to set on a name here, just think we should match behavior between this and https://github.com/kubernetes-sigs/service-apis/pull/268

Reply:
I did consider moving the type into TLSConfig but had similar though process as Rob on this one.
Regarding name of the struct itself, I think RouteTLSConfig makes sense here since we are referring to downstream TLS configuration and I expect that to be same across multiple types of Routes (HTTPRoute or TLSRoute). Thoughts?

Reply:
I'm hugely +1 on establishing a naming pattern for the different TLS config types. I could live with `TLSInRouteConfig` and `TLSInListenerConfig`, etc. Updating that in a separate PR would be best though.

Reply:
Renamed this to RouteTLSConfig.